compose/config/config.py

```diff
@@ -593,7 +593,7 @@ def validate_and_construct_extends(self):
         config_path = self.get_extended_config_path(extends)
         service_name = extends['service']
 
-        if config_path == self.config_file.filename:
+        if config_path == os.path.abspath(self.config_file.filename):
```

Comment:
I think this'd need to be `expand_path(self.working_dir, self.config_file.filename)` instead to correctly handle cases where `working_dir` is not `.`

Reply:
@amezin is right in the sense that `expand_path(self.working_dir, self.config_file.filename)` does not pass the tests.

@rumpl @chris-crone